lib/knowngithub.rb

```diff
@@ -1,5 +1,52 @@
+require 'ipaddr'
+require 'json'
+require 'net/http'
+require 'net/ssh'
+require 'nokogiri'
+require 'open-uri'
 require "knowngithub/version"
 
 module Knowngithub
-  # Your code goes here...
+  def self.fingerprints
+    output = Nokogiri::HTML(open("https://help.github.com/articles/github-s-ssh-key-fingerprints/"))
+    fields = output.xpath("//code")
+    return fields.children.map(&:content)
+  end
+
+  def self.session
+    s = Net::SSH::Transport::Session.new('github.com', { :verify_host_key => true })
+    s.close
+    s
+  end
+
+  def self.host
+    s = self.session
+    if self.fingerprints.any?{|f| f == s.host_keys.first.fingerprint}
+      base64_key = [Net::SSH::Buffer.from(:key, s.host_keys.first).to_s].pack("m*").gsub(/\s/, "")
```

Comment:
My ruby sucks but it appears you aren't validating input (`[0-9A-F:]+`?)

Reply:
@emkll  In this case, `s.host_keys` defers to a bunlded version `OpenSSL::Pkey` within `Net:SSH`, which defers to the wrapped OpenSSL C library. I honestly can't trace where validation is happening at this point, but the object type for `s.host_keys` values is definitely `Net::SSH::HostKeys` which should prevent most evilness.

I'm under the impression [this](https://github.com/net-ssh/net-ssh/blob/master/lib/net/ssh/known_hosts.rb#L43) bit makes sure the keys are proper.

Reply:
Otherwise, everything else you mentioned should be addressed.